File: CPP/linearSearch.cpp
#include<iostream>
using namespace std;

void search(int a[],int N, int n)
{
    int count=0;
    
    for(int i=0;i<N;i++)
    {
        if(a[i]==n)
         {
            cout <<"The number "<<n<<" occured at index"<<i+1<<endl;
            count++;
         }
    }

cout<< " It occured "<<count<<" times \n";
}
int main()
{
    int N;
    int x;
    int *arr = new int[N];
    cout<<"Enter the size of the array and the element to be searched \n";
    cin >>N>>x;
    for(int i=0;i<N;i++)
    {
        cin>>arr[i];
    }
    search(arr,N,x);
    
    delete[] arr;       // To delete dynamically allocated memory for the array arr
    
    return 0;

}

File: PYTHON/linearSearch.py
print('Program for Linear Search')
numbers=[10,23,15,40,43,35,22,60,55,45,70,80,75]#the array of numbers to be searched
print('Enter the number to be searched: ')
num = int(input())#user inputs the number to be searched
loc=-1#location is initally set to -1 to indicate number is not present
flag='is not'#flag is initially set to number not present
for i in range(len(numbers)):#loop over the entire array to see a match
    if numbers[i]==num:#if a match is found
        loc=i+1#add 1 to index as index of array starts at 0
        flag='is'#set flag to number 'is' present
        break#once match found, exit loop
print('The number '+str(num)+' '+flag+' present in the array "numbers"')
print('It is present at location: '+str(i+1))
        